packages/block-library/src/navigation/edit.js

```diff
@@ -126,56 +182,181 @@ function Navigation( {
 	}
 
 	function handleCreateFromExistingPages() {
-		updateNavItemBlocks( defaultPagesNavigationItems );
+		updateNavItemBlocks( navLinkBlocksFromPages );
 		selectBlock( clientId );
 	}
 
-	const hasPages = hasResolvedPages && pages && pages.length;
+	function handleCreateFromExistingMenu() {
+		updateNavItemBlocks( navLinkBlocksFromMenuItems );
+		selectBlock( clientId );
+	}
 
-	const blockInlineStyles = {
-		fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
-	};
+	function handleCreate() {
+		const { key } = selectedDropDownOption;
+
+		// Explicity request to create empty.
+		if ( key === CREATE_EMPTY_OPTION_VALUE ) {
+			return handleCreateEmpty();
+		}
+
+		// Create from Pages.
+		if ( hasPages && key === CREATE_FROM_PAGES_OPTION_VALUE ) {
+			return handleCreateFromExistingPages();
+		}
+
+		// Create from WP Menu (if exists and not empty).
+		if (
+			hasMenus &&
+			selectedDropDownOption &&
+			navLinkBlocksFromMenuItems?.length
+		) {
+			return handleCreateFromExistingMenu();
+		}
+
+		// Default to empty menu
+		return handleCreateEmpty();
+	}
+
+	function buildPlaceHolderInstructionText() {
+		if ( hasMenus && hasPages ) {
+			return __(
+				'Create a Navigation from all existing pages, or chose a Menu.'
+			);
+		}
+
+		if ( ! hasMenus && ! hasPages ) {
+			return __( 'Create a an empty Navigation.' );
+		}
+
+		if ( hasMenus && ! hasPages ) {
+			return __( 'Create a Navigation from a Menu or create empty.' );
+		}
+
+		if ( ! hasMenus && hasPages ) {
+			return __(
+				'Create a Navigation from all existing pages, or create empty.'
+			);
+		}
+	}
+
+	const dropDownOptions = [
+		{
+			id: CREATE_PLACEHOLDER_VALUE,
+			name: __( 'Select where to start from…' ),
```

Comment:
```suggestion
			name: __( 'Select where to start from…' ),
```

This text doesn't feel quite right to me and I'd appreciate a copy/design review or just more eyes on it.

My suggestion would be something like "Choose a starting point", but not sure if that conveys what the menu contains.

Reply:
> I don't really see the copy issues as a blocker (though typos should be fixed), but lets at least make sure a follow-up issue is created.

https://github.com/WordPress/gutenberg/pull/18869#discussion_r435107636

Does this still apply? Shall I make an Issue to track all these copy concerns?

Reply:
Issue tracking copy changes is here https://github.com/WordPress/gutenberg/issues/22987